### DISCUSSION TITLE
Using TUnit as a runner for Playwright? Anything special that should be considered?

### DISCUSSION THREAD

**Discussion by turbobobbytraykov** [thomhurst/TUnit#622] [General]
I know it may be a silly question to ask, but given [Playwright's current documentation on supported test runners](https://playwright.dev/dotnet/docs/test-runners) , I was wondering it the Microsoft Test(ing) Platform (Protocol)/testfx/mttp/mtp or TUnit would expose any official integration with Playwright?

From what I can see, it's that MSTest & NUnit have integration in terms of settings for Playwright's browser instance(s) as well as a PageContext and BrowserContext helper classes.

OFFTOPIC: The "Microsoft Test(ing) Platform (Protocol)" has so many naming variations that it took me a boat-load of time to get lost where to look for details, other than asking the Amaury Leve on Twitter. 😄

**Comment by thomhurst** [comment]
I did raise this on their GitHub but they want the package to be more popular first: https://github.com/microsoft/playwright-dotnet/issues/3007

I could recreate their logic and make my own package though. So I'll keep this issue open and see if I can find time for it.

**Reply by turbobobbytraykov** [reply]
Thanks for the info! Fingers crossed that they do open up to this.
Alternatively, packages or a tutorial from you would suffice for the time being.
That threshold may be 20M for all we know.

**Reply by thomhurst** [reply]
Sorry for the delay. I've mimicked their logic in my own package. If you want to help trial it out, I'm just releasing it to NuGet now.

It's `TUnit.Playwright`

**Reply by turbobobbytraykov** [reply]
Cool! 
Unfortunately, my team has invested several days in nUnit at this point so for now we're set on that course, but I'll keep your change in mind.
However, I've let a friend of mine know so that he can try this out with his colleagues - they have new projects every 6-12 months so they can experiment with tech much more frequently.

**Comment by indcoder** [comment]
@thomhurst Is there any documentation for this TUnit and Playwright? The Web example also does not have it.

**Reply by thomhurst** [reply]
https://thomhurst.github.io/TUnit/docs/examples/playwright

**Comment by mariohik** [comment]
@thomhurst How can I start with headless = false?

**Reply by thomhurst** [reply]
I tend to do this so it's only not headless when I'm debugging

```csharp
[Before(TestSession)]
public void PlaywrightDebug()
{
    if(Debugger.IsAttached)
    {
        Environment.SetEnvironmentVariable("PWDEBUG", "1");
    }
}
```
